com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetGenerator.java

```diff
@@ -374,33 +435,67 @@ public static BufferedImage layoutImages(LayoutResult layoutResult, Map<String,
      * 5. Create vertex data for each frame (image) in each animation
      *
      * @param images list of images
-     * @param imagePaths corresponding image-id to previous list
-     * @param animations list of animations
-     * @param margin internal atlas margin
-     * @return {@link AtlasMap}
+     * @param imageHullSizes image trimming vertex count list
+     * @param imageSourceRects source image information list (offset x, offset y, original width, original height)
+     * @param paths corresponding image-id to previous lists
+     * @param iterator iterator of the list of animations
+     * @param margin internal atlas margin size
+     * @param innerPadding atlas padding size
+     * @param extrudeBorders atlas extrude borders size
+     * @param rotate are images allowed to be rotated
+     * @param useTileGrid when using with tile grids
+     * @param gridSize grid size of the tile grid
+     * @return {@link TextureSetResult}
      */
-    public static TextureSetResult generate(List<BufferedImage> images, List<Integer> imageHullSizes, List<String> paths, AnimIterator iterator,
-            int margin, int innerPadding, int extrudeBorders, boolean rotate, boolean useTileGrid, Grid gridSize) {
+    public static TextureSetResult generate(List<BufferedImage> images, List<Integer> imageHullSizes, List<Point4i> imageSourceRects, List<String> paths, AnimIterator iterator,
```

Comment:
My small comment is that added parameters could be added as last ones, what do you think?